modules/renter/hostdb/hostweight.go

```diff
@@ -322,12 +324,19 @@ func (hdb *HostDB) uptimeAdjustments(entry modules.HostDBEntry) float64 {
 			// Ignore the unsorted scan entry.
 			continue
 		}
+
+		blockDiff := scan.BlockHeight - recentBlockHeight
+		timePassed := float64(time.Duration(blockDiff) * 10 * time.Minute)
+		halftime := float64(1 * time.Hour * 24 * 30)
+		decay := math.Pow(0.5, timePassed/halftime)
+
 		if recentSuccess {
-			uptime += scan.Timestamp.Sub(recentTime)
+			uptime += time.Duration(timePassed * decay)
```

Comment:
This code here should do pretty much the same as the one in `updateEntry`. At a first glance it seems like `uptime` and `downtime` themselves are not decayed.

Reply:
Now that I think about it we can probably also change `uptime` `downtime` `historicUptime` and `historicDowntime` to be of type `blockheight` instead of `Duration` if we decide to do everything using blocks.

When we calculate the ratio later we would just cast them to `float64`.

Reply:
We shouldn't turn `historicUptime` and `historicDowntime` into type `BlockHeight`. 
Since, according to David we should time (and not block height) should be used for things (such as hosts) that lives outside the protocol